### PR TITLE
ENH: ipython syntax

### PR DESCRIPTION
**Pull request by FHaase**
* tests updated as correct line is reported.

### CODE REVIEW

File: flake8_rst/rst.py [hunk 1]
```diff
@@ -20,15 +19,51 @@
...
 )
 
-EXPRESSIONS = (RST_RE, ANCHOR_RE)
+EXPRESSIONS = [RST_RE, ANCHOR_RE]
+
+
+def strip_pycon(src, indent, line_number, *expressions):
+    has_stripped = False
+    for code_block in src.split('\n\n'):
+        matches = (match for expression in expressions for match in re.finditer(expression, code_block))
```
Comment (kataev) [inline flake8_rst/rst.py]:
```
(match for expression in expressions for match in expression.finditer(code_block))
```
?

File: flake8_rst/rst.py [hunk 2]
```diff
@@ -20,15 +19,51 @@
...
+        current_block = []
+        for match in matches:
+            origin_code = match.group('code')
+            indent_ = match.group('indent')
+            output = match.group('output')
+
+            if origin_code is not None:
+                current_block.append((origin_code, len(indent_) if indent_ else 0))
```
Comment (kataev) [inline flake8_rst/rst.py]: Maybe make `indent_ = match.group('indent') or ''` and leave here only `len(indent_)`.
Also why you add _ to indent variable name?

File: flake8_rst/rst.py [hunk 3]
```diff
@@ -37,25 +72,14 @@ def find_sourcecode(src):
...
-                for anchor in ANCHORS:
-                    if line.startswith(anchor):
-                        lines.append(line[len(anchor):])
-                        break
-
-            code = '\n'.join(lines)
+            indent = 0
 
-        line_number = src[:match.start()].count('\n') + match.group('before').count('\n')
+        code = "\n".join([line[indent:] for line in origin_code.split("\n")])
```
Comment (kataev) [inline flake8_rst/rst.py]: if rstrip code here, we can simple return `strip_pycon(code, indent, line_number, INTERN_ANCHOR_RE, ANCHOR_IN_RE)`

File: flake8_rst/rst.py [hunk 4]
```diff
@@ -37,25 +72,14 @@ def find_sourcecode(src):
...
+        start__count = src[:match.start()].count('\n')
```
Comment (kataev) [inline flake8_rst/rst.py]: Why underscore doubled?

**Comment by FHaase** [comment]
Note that I included `.. code-blocks::` in the checks as I'd rather have those code-blocks checked.
If it contains some other language, checks can be disabled by adding that language specifier.

**Comment by FHaase** [comment]
@kataev Sorry, shouldn't have started the PR that early.

Summing all changes up:
* line_numbers and indents all point to the correct values. (except for when In [X]: changes to In[XX]:)
* --show-source option returns correct lines. and ^ points to the correct value. (except for ^ case)
* IPython-Syntax gets checked only when in .. code-blocks::
* If file ends with *.py(*) only docstrings are checked.

**Comment by kataev** [comment]
I’m now at conference, i will answer little later.

**Comment by kataev** [comment]
Ok, nice. Thanks.

**Pull request merged**